### ISSUE TITLE
Icon Notes not available on some list cards : ThirdParty, Project, Shipment,Products...

### ISSUE THREAD

**Issue by josett225** [Dolibarr/dolibarr#13645]
# Instructions
*This is a template to help you report good issues. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *replace the bracket enclosed texts with meaningful information*
- *remove any unused sub-section*

# Bug
[Icon Notes missing in several list cards : Shipping, Project, ThirdParty, Products, Services,...]

## Environment
Version: [11.0.3] and [12.x]
OS: [Linux]
Web server: [Apache/2.2.34 (Unix)]
PHP: [7.4.2]
Database: [MYSqli 5.7.26]
URL(s):[/projet/list.php?leftmenu=projets;/expedition/list.php?leftmenu=sendings;/societe/list.php?leftmenu=thirdparties]
﻿﻿

## Expected and actual behavior
[The Notes Icon appears in several list cards as Proposal, Order, Invoices as tested. See Screenshots. In Project, Shipment, Products, Thirdparty, the icon doesn't appear. This is great tool to get immediate information to manage projects, shipments whithout the need to discover if there are some notes attached and to get immediate access to the information. The behavior expected is as Proposal, Orders and Invoices. ]

## Steps to reproduce the behavior
[Please go to a shipment and/or a project card, select Note Tab and then type a private note. Please go to the list and you will not see the icon on the shipment or project list]

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)

[<img width="296" alt="Capture d'écran 2020-04-17 18 30 07" src="https://user-images.githubusercontent.com/57623859/79595538-5330f080-80df-11ea-8a52-52f35a172828.png">
<img width="397" alt="Capture d'écran 2020-04-17 18 30 19" src="https://user-images.githubusercontent.com/57623859/79595542-54621d80-80df-11ea-88ad-460c254a3f45.png">
<img width="203" alt="Capture d'écran 2020-04-17 18 30 30" src="https://user-images.githubusercontent.com/57623859/79595545-55934a80-80df-11ea-94d5-0219d6a27d8a.png">
<img width="258" alt="Capture d'écran 2020-04-17 18 30 41" src="https://user-images.githubusercontent.com/57623859/79595547-562be100-80df-11ea-9dbb-65f6a2f29780.png">
<img width="325" alt="Capture d'écran 2020-04-17 18 31 18" src="https://user-images.githubusercontent.com/57623859/79595548-56c47780-80df-11ea-82c6-5f17cc8bef7a.png">
<img width="202" alt="Capture d'écran 2020-04-17 18 44 26" src="https://user-images.githubusercontent.com/57623859/79595552-588e3b00-80df-11ea-865d-7c77388258ae.png">
]

**Comment by atm-maxime** [comment]
Yes you're right, this icon has been added on some lists but not all for the moment. This is not a bug but a feature request to continue adding this on other lists.

**Comment by josett225** [comment]
Hi @atm-maxime thanks for answering. Is there any chance to get it done in V12 as in a usability and time adoption end user perspective, to get it in project/opportunity and shipment lists, its critical business requirement. There are more than 200 opportunities and 200 shipments in a flow window. Maybe we are wrong in using the following user behavior process. Status and Icons with particularly Notes icon (MAIN_AUTO_TIMESTAMP_IN_PRIVATE_NOTES activated : Time Stamp and User Stamp) are used with different usersgroup to run the full process from opportunities/Proposals/orders/shipments/invoices.   So small things can make a huge business impact.

**Comment by atm-maxime** [comment]
Hi !
V12 is already frozen and the beta just started, so there will be no new feature. And to "get it done", you need a developer to volunteer for the task, open-source project is based on volonteering, so there's no expectations to have. Or you can fund a developer to do it and share with the community.
As for your own process, I couldn't tell if it's appropriate without knowing it, but maybe you can check on extrafields to store data on your business documents and use them on lists with filters.
Cheers :)

**Comment by josett225** [comment]
@atm-maxime Thanks for your return. Yes I perfectly understand your opensource volunteer explanation as I already funded several OpenSource projects/companies. I am ready to evaluate funding if I know either who has the skills on this specific item (don't want to lose funding on reverse engineering time) or if we could fund the core team association or if the process is explained somewhere to make it develop internally. How to be sure this development will be added in the core (after review and validation of course) in the future as I don't want to get specific code to be maintained and then struggle each time a new Dolibarr release is coming out? You raised a beta is coming soon. where could we find the beta information? As a enduser and admin of the solution, is there any usergroup/conference where we could attend virtually to share inputs, feedback, get info or requests? Be part of the association could help?

**Comment by atm-maxime** [comment]
@josett225 Any active developer in the community would have the skills to do this task. The Dolibarr fundation is not concerned by technical developments, so if you want this to be developed and added to the core you should contact one of the Dolibarr partners. They're listed in the Dolibarr wiki.
Information about new releases and beta are shared on Dolibarr websites, forums, social medias, ... (https://www.dolibarr.org/forum/). These are the places you can find many informations and share with others.

**Comment by atm-maxime** [comment]
Still relevant

**Comment by josett225** [comment]
Still required 

Jose MARTINEZ
M: +33 6 14 18 69 22

**Comment by josett225** [comment]
Still required.

**Comment by josett225** [comment]
Hi @eldy and @atm-maxime

After 3 years learning Dolibarr with both using it internally and deploying it in several customers thru Pichinov based in Toulouse and after some PRs, I started the project to add notes in all objects list.
I would like your advise before sending all the PRs.
I divided the project in 2 phases :
- Add both public notes and private notes in actual objects. I changed the color of private picto in black and let the public note in grey (as today)
Here is the first example with Order list.
![CommandeNotesGlobal](https://github.com/Dolibarr/dolibarr/assets/57623859/93a91671-bdca-4d93-9897-bc328249e8df)
![CommandeNotesPrivate](https://github.com/Dolibarr/dolibarr/assets/57623859/c49c6d3d-a7cb-403d-9a08-1c04171451b8)
![CommandeNotesPublic](https://github.com/Dolibarr/dolibarr/assets/57623859/159c82a7-6356-46f3-88c0-8fee0e24f3d2)

- Add same structure in all objects without notes
Here is an example with products list :
![ProductNoteGlobal](https://github.com/Dolibarr/dolibarr/assets/57623859/7a56138d-85e2-4eae-8970-edb8c4de7b9d)
![ProductNotePublic](https://github.com/Dolibarr/dolibarr/assets/57623859/d2e82886-1d88-4ace-ba53-2970849d7cf4)
![ProductNotePrivate](https://github.com/Dolibarr/dolibarr/assets/57623859/0e4879f6-578f-404a-9e5e-05e699407955)



Could you please comment or amend the project?

Best Regards

Jose

**Comment by atm-maxime** [comment]
Hi Jose !
My point of view on this subject is that we should avoid adding too many pictos on lists. Maybe only one picto as it is now but displaying both public and private note in the tooltip is a good solution?
Happy to hear you have achieved work on this matter to share with the community!

**Comment by josett225** [comment]
Hi @atm-maxime 

Thanks a lot for the feedback, I will work on it based on your comment and 2 ideas I have to keep it clear between public only, private only or both.

**Comment by josett225** [comment]
Still in progress

**Comment by josett225** [comment]
Now I have a team, we will finish it.

**Comment by frederic34** [comment]
Two follow-ups on develop:

- #40614 adds optional "Note (public)"/"Note (private)" list columns to project and product/service lists (shipment and thirdparty lists already got this at some point since this issue was filed).
- #40615 adds the actual mechanism this issue describes — the small clickable note picto shown right after the ref/name link (already present on proposal/order/invoice lists via `$addlinktonotes` in `getNomUrl()`) — to project, product, shipment and thirdparty lists too.
